<launch>   

  <node name="urg_node" pkg="urg_node" type="urg_node" output="screen">
    <param name="ip_address" value="192.168.0.10"/>
    <param name="serial_port" value="/dev/ttyACM0"/>
    <param name="serial_baud" value="115200"/>
    <param name="frame_id" value="base_link"/>
    <param name="calibrate_time" value="true"/>
    <param name="publish_intensity" value="false"/>
    <param name="publish_multiecho" value="false"/>
    <param name="angle_min" value="-3.141592654"/>
    <param name="angle_max" value="3.141592654"/>
  </node>

  <!-- start PointCloud Fast Filter -->
  <node name="velodyne" pkg="pff" type="velodyne">
    <param name="robot_height" value="2.0"/> <!-- Default 1.0 (100cm) -->
    <param name="sensor_height" value="0.39"/> <!-- Default 0.57 (57cm) -->
    <param name="horizontal_fov" value="180"/> <!-- Default 360° -->
    <param name="resolution" value="1"/> <!-- Default 0.2° -->
    <param name="topic_pub" value="/scan"/> <!-- Default /scan -->
    <param name="topic_sub" value="/velodyne_points"/> <!-- Default velodyne_points -->
    <param name="frame_id" value="base_link"/> <!-- Default base_scan -->
  </node> 

  <!-- start Turtlebot -->
  <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>

  <!-- start gmapping -->
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping">
    <remap from="scan" to="scan"/>
    <param name="map_update_interval" value="5.0"/>      
    <param name="linearUpdate" value="0.0"/>
    <param name="angularUpdate" value="0.0"/>
  </node>	


  <!-- start rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pff)/rviz_cfg/slam.rviz" />

</launch>
